### PR TITLE
Optimizes getviewsize

### PR DESCRIPTION
**Pull request by deathride58**
:cl: deathride58
code: Improved the performance of getviewsize (feat. code stolen from ninjanomnom's review)
/:cl:

I have no idea how to profile code, so I have no idea exactly how much this improves getviewsize's performance.

### CODE REVIEW

**Comment by Cyberboss** [comment]
You've essentially just reorganized the code I think

File: code/__HELPERS/view.dm [hunk 1]
```diff
@@ -1,12 +1,7 @@
...
-		viewY = totalviewrange
-	else
-		var/list/viewrangelist = splittext(view,"x")
-		viewX = text2num(viewrangelist[1])
-		viewY = text2num(viewrangelist[2])
-	return list(viewX, viewY)
\ No newline at end of file
+		. = 1 + 2 * view
+		return list(.,.)
+	if(istext(view))
```
Comment (Cyberboss) [inline code/__HELPERS/view.dm]: I suppose the istext makes a difference though

**Comment by deathride58** [comment]
Apparently defining vars in a proc harms performance, if I read last night's discussion on the IRC revolving around this topic right.

In theory, making use of . instead of defining a bunch of different variables in proc should result in slightly improved performance. If I knew how to profile procs, I would be able to test the performance improvements myself.

**Comment by Cyberboss** [comment]
Well, they were talking about atmos in that context. How often is this proc called?

**Comment by deathride58** [comment]
It's called whenever a click catcher or parallax layer gets updated, it's called every time can_use_topic is called for silicons and ghosts, and it's called every time a mob fires a pixel projectile. Performance gains would be miniscule, but a half femtosecond improvement's a half femtosecond improvement, gdi

**Comment by kevinz000** [comment]
*sighs*

File: code/__HELPERS/view.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/__HELPERS/view.dm]: | Proc Name | Self CPU | Total CPU | Real Time | Calls |
| --- | --- | --- | --- | --- |
| /proc/new_viewsize | 0.358 | 0.360 | 0.368 | 100000 |
| /proc/old_viewsize | 0.337 | 0.338 | 0.347 | 100000 |
| /proc/my_viewsize | 0.256 | 0.257 | 0.269 | 100000 |

This istext is the problem, it was just an else before.

Also this proc could be made faster anyway

```
/proc/getviewsize(view)
	if(length(view))
		return splittext(view, "x")
	else
		var/num = 1 + 2 * view
		return list(num, num)
```

**Review by ninjanomnom — DISMISSED** [review]

File: code/__HELPERS/view.dm [hunk 2]
```diff
@@ -1,12 +1,6 @@
...
+		return splittext(view,"x")
 	else
-		var/list/viewrangelist = splittext(view,"x")
-		viewX = text2num(viewrangelist[1])
-		viewY = text2num(viewrangelist[2])
-	return list(viewX, viewY)
\ No newline at end of file
+		var/num = 1+2*view
+		return list(num,num)
```
Comment (Cyberboss) [inline code/__HELPERS/view.dm]: The else isn't needed

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/__HELPERS/view.dm [hunk 3]
```diff
@@ -1,12 +1,6 @@
 /proc/getviewsize(view)
-	var/viewX
-	var/viewY
-	if(isnum(view))
-		var/totalviewrange = 1 + 2 * view
-		viewX = totalviewrange
-		viewY = totalviewrange
+	if(istext(view))
+		return splittext(view,"x")
```
Comment (AnturK) [inline code/__HELPERS/view.dm]: Doesn't this still need the num2text

**Review by AnturK — CHANGES_REQUESTED** [review]

**Comment by duncathan** [comment]
stale

**Pull request closed without merging**